writer.go

```diff
@@ -81,5 +81,5 @@ func CombineWriteSyncers(writers ...zapcore.WriteSyncer) zapcore.WriteSyncer {
 	if len(writers) == 0 {
 		return zapcore.AddSync(ioutil.Discard)
 	}
-	return zapcore.Lock(zapcore.NewMultiWriteSyncer(writers...))
```

Comment:
I'd prefer to keep this auto-lock, since it lets users easily replicate the behavior of the `Config` struct. In general, I'd like the `zapcore` package to have no defaults and the `zap` package to supply sane defaults.

Anyone who doesn't want this auto-locking can just use `zapcore.NewMultiWriteSyncer`.